vault/logical_system.go

```diff
@@ -725,6 +729,140 @@ func (b *SystemBackend) handlePluginReloadUpdate(ctx context.Context, req *logic
 	return &r, nil
 }
 
+func (b *SystemBackend) handlePluginRuntimeCatalogUpdate(ctx context.Context, _ *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+	runtimeName := d.Get("name").(string)
+	if runtimeName == "" {
+		return logical.ErrorResponse("missing plugin runtime name"), nil
+	}
+
+	runtimeTypeStr := d.Get("type").(string)
+	if runtimeTypeStr == "" {
+		return logical.ErrorResponse("missing plugin runtime type"), nil
+	}
+
+	runtimeType, err := consts.ParsePluginRuntimeType(runtimeTypeStr)
+	if err != nil {
+		return logical.ErrorResponse(err.Error()), nil
+	}
+
+	switch runtimeType {
+	case consts.PluginRuntimeTypeContainer:
+		ociRuntime := d.Get("oci_runtime").(string)
+		if ociRuntime == "" {
```

Comment:
I think we will have an easier time supporting this API in the vault terraform provider if we faithfully represent the data passed to us in storage, and set up defaults when starting a plugin rather than when storing the runtime config.

Reply:
Good point!